openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,71 @@
+""" Management command to update is_prerequisite_courses_passed flag in ApplicationHub """
+import logging
+import sys
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from lms.djangoapps.courseware.courses import get_course
+from lms.djangoapps.grades.api import CourseGradeFactory
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.adg.lms.course_meta.models import CourseMeta
+from student.models import CourseEnrollment
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Updates is_prerequisite_courses_passed flag in ApplicationHub.
+    """
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        pre_req_course_metas = CourseMeta.pre_reqs.all()
+
+        if not pre_req_course_metas:
+            sys.exit('Exiting!!! No pre-req courses found. But there must be some pre-reqs. Please add in admin.')
+
+        users_need_to_be_updated = self.get_users_need_to_be_updated(pre_req_course_metas)
+        pre_req_courses = [get_course(course_meta.course.id) for course_meta in pre_req_course_metas]
+
+        for user in users_need_to_be_updated:
+            for course in pre_req_courses:
+                if self.is_user_failed_in_course(user, course):
+                    logger.info('{username} has not yet passed all the pre-reqs'.format(username=user.username))
+                    break
+            else:
+                self.mark_pre_req_courses_as_passed(user)
+
+    def get_users_need_to_be_updated(self, pre_req_course_metas):
+        """
+        1. Gets user enrolled in first pre-req course, Because any user which is not enrolled
+        in any pre-req doesn't deserve an update. So optimistically we can only check all
+        users enrolled in any, one pre-req. Here we have chosen the first pre-req.
+        2. From user enrolled in first pre-req, filter those whose application Hub doesn't
+        exist OR is_prerequisite_courses_passed flag is false.
+        """
+        users_enrolled_in_first_pre_req = CourseEnrollment.objects.users_enrolled_in(
```

Comment:
Why are we checking only first pre-req?

Reply:
Yes, this is the best optimal way to get users on which we should iterate. Why is it best optimal way, please have a look into point 1 of `doc string`

Reply:
But still we will be updating the status of those users who have completed all pre-req courses? don't we?

Reply:
Yes, status is completed of those users who have completed all.